include/zephyr/arch/xtensa/xtensa_mmu.h

```diff
@@ -9,8 +9,47 @@
 
 #define Z_XTENSA_MMU_X  BIT(0)
 #define Z_XTENSA_MMU_W  BIT(1)
+#define Z_XTENSA_MMU_XW  (BIT(1) | BIT(0))
+
+#ifndef CONFIG_XTENSA_MMU_WA_NO_PTE_CACHE
 #define Z_XTENSA_MMU_CACHED_WB BIT(2)
 #define Z_XTENSA_MMU_CACHED_WT BIT(3)
+#else
+#define Z_XTENSA_MMU_CACHED_WB 0
+#define Z_XTENSA_MMU_CACHED_WT 0
+#endif
+
+#define K_MEM_PARTITION_IS_EXECUTABLE(attr)	(((attr) & Z_XTENSA_MMU_X) != 0)
+#define K_MEM_PARTITION_IS_WRITABLE(attr)	(((attr) & Z_XENSA_MMU_W) != 0)
+
+/* Read-Write access permission attributes */
+#define K_MEM_PARTITION_P_RW_U_RW ((k_mem_partition_attr_t) \
+			{Z_XTENSA_MMU_W})
+#define K_MEM_PARTITION_P_RW_U_NA ((k_mem_partition_attr_t) \
+			{0})
+#define K_MEM_PARTITION_P_RO_U_RO ((k_mem_partition_attr_t) \
+			{0})
+#define K_MEM_PARTITION_P_RO_U_NA ((k_mem_partition_attr_t) \
+			{0})
+#define K_MEM_PARTITION_P_NA_U_NA ((k_mem_partition_attr_t) \
+	{0})
+
+/* Execution-allowed attributes */
+#define K_MEM_PARTITION_P_RWX_U_RWX ((k_mem_partition_attr_t) \
+			{Z_XTENSA_MMU_XW})
```

Comment:
FWIW: I discovered while looking at MPU support recently that we have (AFAICT) no test coverage of the RWX attributes.  I'd suggest removing this.  If we never promised that it worked, we should pick a W^X architecture from the start.  It's a useful and simple security paradigm.